File: apps/hello_micromag.cpp
#include <iostream>
#include "micromag/grid.hpp"
#include "micromag/field.hpp"
#include "micromag/vtk_writer.hpp"

int main() {
    using namespace micromag;

    // 64 nm x 64 nm x 8 nm, 2 nm cells -> 32 x 32 x 4
    StructuredGrid grid(32, 32, 4, 2e-9, 2e-9, 2e-9);
    VectorField3D m(grid);

    Vec3 ext = grid.extent();
    m.set_vortex(ext.x * 0.5, ext.y * 0.5, /*core_radius=*/8e-9);

    write_vtk_legacy("vortex.vtk", m);

    std::cout << "Wrote vortex.vtk: "
              << grid.nx() << " x " << grid.ny() << " x " << grid.nz()
              << " cells (" << grid.size() << " total)\n";
    std::cout << "Open in ParaView to visualize.\n";
    return 0;
}

File: examples/hello_micromag.py
"""Phase 1a demo: create a vortex magnetization and save to VTK."""
import os, sys
from pathlib import Path

import sys
from pathlib import Path
sys.path.insert(0, str(Path(__file__).resolve().parent.parent))
from micromag_locate import add_micromag_to_path
add_micromag_to_path()
import micromag as mm

def main():
    grid = mm.StructuredGrid(nx=32, ny=32, nz=4,
                             dx=2e-9, dy=2e-9, dz=2e-9)
    field = mm.VectorField3D(grid)

    ext_x = grid.nx * grid.dx
    ext_y = grid.ny * grid.dy
    field.set_vortex(cx=ext_x * 0.5, cy=ext_y * 0.5, core_radius=8e-9)

    mm.write_vtk_legacy("vortex_py.vtk", field, "m")
    print(f"Wrote vortex_py.vtk: {grid.nx} x {grid.ny} x {grid.nz} "
          f"cells ({grid.size} total)")

if __name__ == "__main__":
    main()
